app/controllers/users.js

```diff
@@ -96,14 +97,96 @@ exports.invitePlayers = (req, res) => {
 
     // send mail with defined transport object
     transporter.sendMail(mailOptions, (error, info) => {
-      if (error) {
-        return res.json(error);
+      if (res) {
+        if (error) {
+          return res.json(error);
+        }
+        return res.json({ sent: true, info });
       }
-      return res.json({ sent: true, info });
     });
   });
 };
 
+const sendInviteNotification = (recieverEmail, req, res) => {
+  const details = {
+    url: req.body.urlLink,
+    receiver: recieverEmail,
+    sender: req.decoded.email,
+    message: `You have 
+      been invited to join a game by ${req.decoded.name}, 
+      click this message to join ${req.body.urlLink}`
+  };
+  Notification.create(details)
+    .then((notification) => {
+      res.status(201).send({
+        success: true,
+        message: notification.message,
+        notification
+      });
+    })
+    .catch((error) => {
+      res.status(400).send({
+        success: false,
+        message: 'Error occurred while sending notification(s)',
+        error: error.message
+      });
+    });
+};
+
+exports.invitePlayers = (req, res) => {
+  const recieverEmail = req.body.email;
+  User.findOne({
+    email: recieverEmail
+  }).exec((err, user) => {
+    if (err) {
+      res.status(500).send({
+        success: false,
+        message: 'Internal server error'
+      });
+    }
+    if (!user) {
+      // invite by email
+      sendInviteMail(recieverEmail, req, res);
+    } else {
+      // invite by notification and email
+      sendInviteMail(recieverEmail, req);
+      sendInviteNotification(recieverEmail, req, res);
+    }
+  });
+};
+
+exports.getGameInviteNotification = (req, res) => {
+  Notification.find({
+    receiver: req.decoded.email,
+    read: false
+  })
+    .then((notifications) => {
+      res.status(200).send({
+        success: true,
+        notifications
+      });
+    });
+};
+
+exports.readGameInviteNotification = (req, res) => {
+  Notification.update({
+    receiver: req.decoded.email,
+  }, { read: true, read_at: Date.now() }, { multi: true })
+    .then((response) => {
+      res.status(200).send({
+        success: true,
+        message: 'You have successfully read your notification',
+        response
+      });
+    })
+    .catch((error) => {
+      res.status(400).send({
+        success: false,
+        message: 'Error occurred while fetching your notification',
```

Comment:
Hi, @llabake.
I don't think the error message `Error occurred while fetching your notification` describes the actual error that can occur in this block of code.
The error message is supposed to describe a situation when there is an error reading the **Game Invite Notifications**, but it is describing a situation when there is an error **Fetching the Notifications**. 
Please make the error message more descriptive.